quickwit-indexing/src/split_store/local_split_store.rs

```
@@ -0,0 +1,213 @@
+    pub fn open(
+        local_storage_root: PathBuf,
+        params: IndexingSplitStoreParams,
+    ) -> StorageResult<LocalSplitStore> {
+        let mut cache_entries = HashMap::new();
+        let mut total_size_in_bytes: usize = 0;
+        for dir_entry_result in fs::read_dir(&local_storage_root)? {
+            let dir_entry = dir_entry_result?;
+            if let Some(split_id) = split_id_from_filename(&dir_entry) {
+                let split_num_bytes = dir_entry.metadata()?.len() as usize;
+                total_size_in_bytes += split_num_bytes;
+                cache_entries.insert(split_id, split_num_bytes);
+            }
+        }
+
+        let local_storage = LocalFileStorage::from(local_storage_root);
+
+        if cache_entries.len() > params.max_num_splits {
+            return Err(StorageErrorKind::InternalError.with_error(anyhow::anyhow!(
+                "Initial number of files exceeds the maximum number of files allowed.",
+            )));
+        }
+
+        if total_size_in_bytes > params.max_num_bytes {
+            return Err(StorageErrorKind::InternalError.with_error(anyhow::anyhow!(
+                "Initial cache size exceeds the maximum size in bytes allowed.",
+            )));
+        }
+
+        Ok(LocalSplitStore {
+            local_storage,
+            stored_splits: cache_entries,
+            params,
+        })
+    }
+
+    #[cfg(test)]
+    pub fn inspect(&self) -> &HashMap<String, usize> {
+        &self.stored_splits
+    }
+
+    pub async fn remove_split(&mut self, split_id: &str) -> StorageResult<()> {
+        if !self.stored_splits.contains_key(split_id) {}
+        let split_path = PathBuf::from(quickwit_common::split_file(split_id));
+        if let Err(error) = self.local_storage.delete(&split_path).await {
+            warn!(file_path = %split_path.display(), error = %error, "Could not remove file from local storage.");
```

hmmm we should probably return an error here too.

---

I thought we wanted to be less strict on delete. but yes if we miss this delete, the file might never get a delete request anymore and even if we restart it will still be loaded